quadratic_equation.h

```diff
@@ -3,57 +3,5 @@
 #include "assert.h"
 #define EXP 1.0e-10

```

Comment:
1) No defines!!! 
2) Use C++ compiler (and ofc rename quadratic_equation.c into quadratic_equation.cpp), this will allow you to use some basic features of C++ - for example, using constants and enum values in switch-case.
3) EXP - not self-descriptive name! I would have absolutely no idea what this constant means without looking at where and what for it's used.
4) You can use standard DBL_EPSILON.